odpc.client/src/features/publicatie/composables/use-documenten.ts

```diff
@@ -1,64 +1,30 @@
-import { ref, onMounted, watch, type ComputedRef } from "vue";
+import { ref, onMounted, type ComputedRef, watchEffect } from "vue";
 import { useFetchApi } from "@/api/use-fetch-api";
+import { useAllPages } from "@/composables/use-all-pages";
 import toast from "@/stores/toast";
-import { mimeTypesMap, uploadFile } from "../service";
-import { PublicatieStatus, type PublicatieDocument } from "../types";
+import { uploadFile } from "../service";
+import type { PublicatieDocument } from "../types";
 
 const API_URL = `/api/v1`;
 
 export const useDocumenten = (pubUUID: ComputedRef<string | undefined>) => {
-  const getInitialDocument = (): PublicatieDocument => ({
-    publicatie: "",
-    officieleTitel: "",
-    verkorteTitel: "",
-    omschrijving: "",
-    creatiedatum: new Date().toISOString().split("T")[0],
-    bestandsnaam: "",
-    bestandsformaat: "",
-    bestandsomvang: 0,
-    bestandsdelen: [],
-    status: PublicatieStatus.gepubliceerd
-  });
-
   // Documenten
-
+  const files = ref<File[]>([]);
   const documenten = ref<PublicatieDocument[]>([]);
-  const files = ref<File[]>();
-
-  const {
-    get: getDocumenten,
-    data: documentenData,
-    isFetching: loadingDocumenten,
-    error: documentenError
-  } = useFetchApi(() => `${API_URL}/documenten/?publicatie=${pubUUID.value}`, {
-    immediate: false
-  }).json<PublicatieDocument[]>();
-
-  watch(documentenData, (value) => (documenten.value = value || documenten.value));
 
-  watch(files, () => addDocumenten());
+  const loadingDocumenten = ref(false);
+  const documentenError = ref(false);
 
-  const addDocumenten = () => {
-    const docs: PublicatieDocument[] = [];
+  const getDocumenten = () => {
+    const { data, loading, error } = useAllPages<PublicatieDocument>(
```

Comment:
Ik denk dat we hier niet helemaal de juiste abstracties gebruiken. Ik vind een watchEffect binnen de body van een functie niet zo fraai. Een paar oplossingsrichtingen: 
- fetchAllPages exporteren en hier rechtstreeks gebruiken
- useAllPages slimmer maken mbv [`MaybeRefOrGetter<T>`](https://vuejs.org/api/utility-types#maybereforgetter) en [`toValue()`](https://vuejs.org/api/reactivity-utilities#tovalue), zodat je ofwel een statische url, ofwel een reactive url kan meegeven. Dan kan je dit direct in de body van de `useDocumenten` functie doen en kan de `onMounted` weg. Dit is denk ik meer werk, kunnen we morgen over overleggen.

Reply:
Wordt alleen vanuit onMounted aangeroepen, maar inderdaad niet zo fraai. Prima, morgen even overleggen

Reply:
`fetchAllPages` exported en gebruikt